models/kernels/reservoir/LinearReservoir.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef LINEAR_RESERVOIR_H
+#define LINEAR_RESERVOIR_H
+
+class LinearReservoir
+{
+    public:
+        LinearReservoir(double init_storage = 1.0, double res_max_storage = 2.0, double res_constant = 0.1,
+                double internal_time_step = 3600.0 );
+
+        void update_storage(double input_storage);
+
+        double flux(double dt_secs);
```

Comment:
Looking at LinearReservoir and Nonlinear_Reservoir, it would be a good idea to unify the interfaces (i.e. `flux` and `response_meters_per_second`).  A common abstract interface could seed both classes, and LinearReservoir should use reservoir should use state and param structs (could be inherited from the base Reservoir concept.). This doesn't have to go in this PR, it can be a follow on to this work.

Reply:
It does make sense to combine those into a single inheritance tree.  For various reasons, I just didn't address it at this point, but I agree that should be done.

Reply:
I will work on this.